presto-main/src/main/java/com/facebook/presto/type/CharOperators.java

```diff
@@ -101,6 +102,14 @@ public static long hashCode(@SqlType("char(x)") Slice value)
         return XxHash64.hash(value);
     }
 
+    @LiteralParameters("x")
+    @ScalarOperator(XX_HASH_64)
+    @SqlType(StandardTypes.BIGINT)
+    public static long xxHash64(@SqlType("char(x)") Slice slice)
```

Comment:
This was needed for the approx_distinct. 

Actually the equals semantics for CHAR in Presto are not correct. Because of the binary comparison, the `EQUALS` operator would return false for `abc  ` and `abc`. That will result into storing wrong statistics for the CHAR type. For now i'm not going to collect the NDV statistic for the `CHAR` type, untils the CHAR semantics are fixed.

Reply:
UPD: It is not possible to save CHAR statistics without setting NDV. Going to go with the `return XxHash64.hash(slice);` implementation. But potentially it may return higher number of distinct values.

Reply:
https://github.com/prestodb/presto/pull/11101

Reply:
> Actually the equals semantics for CHAR in Presto are not correct. Because of the binary comparison, the `EQUALS` operator would return false for `abc⎵⎵` and `abc`. 

as noted https://github.com/prestodb/presto/pull/11101#issuecomment-406687814, internal repr is normalized, with trailing spaces removed (wheever there is a trailing space, this is a bug)